pulsar-client/src/main/java/org/apache/pulsar/client/impl/TypedMessageBuilderImpl.java

```diff
@@ -45,10 +46,10 @@ public class TypedMessageBuilderImpl<T> implements TypedMessageBuilder<T> {
 
     private static final ByteBuffer EMPTY_CONTENT = ByteBuffer.allocate(0);
 
-    private final transient ProducerBase<?> producer;
-    private final transient MessageMetadata msgMetadata = new MessageMetadata();
-    private final transient Schema<T> schema;
-    private transient ByteBuffer content;
+    protected final transient ProducerBase<?> producer;
+    protected final transient MessageMetadata msgMetadata = new MessageMetadata();
+    protected final transient Schema<T> schema;
+    protected transient ByteBuffer content;
```

Comment:
You don't need to change these access modifiers, as well as the `beforeSend` method.

You can reuse the `getMessage` method when you want to mock it in `MsgPayloadTouchableMessageBuilder`.

```java
        @Override
        public Message<T> getMessage() {
            final var msg = (MessageImpl<T>) super.getMessage();
            payload = msg.getPayload().retain();
            if (msgMocker == null) {
                return msg;
            } else {
                return msgMocker.apply(msg);
            }
        }
```

Reply:
Changed